### PR TITLE
Result inductor

### PR DESCRIPTION
**Pull request by exclamaforte**
phew

### CODE REVIEW

File: 2.7.0/done/result_inductor.md [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+ - feature_use: Remove JK from naming for feature use. ([#143529](https://github.com/pytorch/pytorch/pull/143529))
+ - Add instantiation level to CutlassArgs ([#144506](https://github.com/pytorch/pytorch/pull/144506))
+ - ROCM hackfix for segfault in addmm op ([#144519](https://github.com/pytorch/pytorch/pull/144519))
+ - Speedup maximum/minumum ops ([#144581](https://github.com/pytorch/pytorch/pull/144581))
+ - Restrict ND tiling analysis to MemoryDeps ([#144497](https://github.com/pytorch/pytorch/pull/144497))
+ - Add support for trunc(). ([#144629](https://github.com/pytorch/pytorch/pull/144629))
+ - Properly generate index expressions ([#144632](https://github.com/pytorch/pytorch/pull/144632))
+ - Implement bitcasts ([#144638](https://github.com/pytorch/pytorch/pull/144638))
+ - Disable scuba logging for autotuning ([#144568](https://github.com/pytorch/pytorch/pull/144568))
+ - - Implement `check_bounds` ([#144635](https://github.com/pytorch/pytorch/pull/144635))
```
Comment (janeyx99) [inline 2.7.0/done/result_inductor.md]: Everything after this has double bullets—unintentional?

File: 2.7.0/done/result_inductor.md [hunk 2]
```diff
@@ -0,0 +1,458 @@
+## inductor
+### bc breaking
```
Comment (janeyx99) [inline 2.7.0/done/result_inductor.md:3]: For bc breaking and deprecation, can you include a 2.6.0 vs 2.7.0 before and after? Please check the guidance doc in the instructions for what is expected for these sections

File: 2.7.0/done/result_inductor.md [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+- Rename `use_absolute_path` to `use_relative_path` in AOTI. This reflects the option's true purpose: compile a cpp file using its basename instead of the its full path ([#147805](https://github.com/pytorch/pytorch/pull/147805)).
+### deprecations
+ - Drop support for Triton versions without ASTSource (around Triton version 2.2.0) ([#143817](https://github.com/pytorch/pytorch/pull/143817)).
+### new features
+ - Config Fuzzer: This is a new debugging tool designed to fuzz Torch compile configurations. Given a test function, it will identify combinations of configs that throw errors during compilation and execution ([#139736](https://github.com/pytorch/pytorch/pull/139736)) ([#145565](https://github.com/pytorch/pytorch/pull/145565)).
+ - Support fusion of pointwise ops into Template Prologs. Setting `TORCHINDUCTOR_PROLOGUE_FUSION` enables this feature ([#147008](https://github.com/pytorch/pytorch/pull/147008)).
+ - Add instantiation level for generating configs in the CUTLASS backend. Set `TORCHINDUCTOR_CUTLASS_INSTANTIATION_LEVEL`. See config docstring for information ([#146230](https://github.com/pytorch/pytorch/pull/146230)).
```
Comment (janeyx99) [inline 2.7.0/done/result_inductor.md]: Nice

**Review by janeyx99 — COMMENTED** [review]
Thanks! This was certainly a lot! Left some comments, one other overarching thing is that I noticed you already started doing a great job categorizing concepts together; more of that with AOTI/flex attn would be more readable too. If you had to prioritize please address the BC breaking/deprecation sections

File: 2.7.0/done/result_inductor.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+class MyPrintObserver(GraphTransformObserver):
+    def get_node_creation_hook(self):
+        def hook(node: torch.fx.Node):
+            print(node)
+        return hook
+```
+
+### deprecations
+
+ - Drop support for Triton versions without ASTSource (around Triton version 2.2.0) ([#143817](https://github.com/pytorch/pytorch/pull/143817)).
```
Comment (janeyx99) [inline 2.7.0/done/result_inductor.md:23]: Does this mean any Triton version 2.2.0 and lower will not be supported anymore? With PyTorch in general?

File: 2.7.0/done/result_inductor.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+### new features
+
+ - Enable non power-of-2 `head_dim` for FlexAttention ([#133495](https://github.com/pytorch/pytorch/pull/133495)).
+ - Add FlexAttention kernel parameter tuning options: `num_warps` and `num_stages` ([#139639](https://github.com/pytorch/pytorch/pull/139639)).
+ - Support vectorization for score and mask in FlexAttention CPU ([#143638](https://github.com/pytorch/pytorch/pull/143638)).
+ - `ConfigFuzzer`: a new debugging tool designed to fuzz Torch compile configurations. Given a test function, it will identify combinations of configs that throw errors during compilation and execution ([#139736](https://github.com/pytorch/pytorch/pull/139736)) ([#145565](https://github.com/pytorch/pytorch/pull/145565)).
+ - Support fusion of pointwise ops into Template Prologs. `TORCHINDUCTOR_PROLOGUE_FUSION` enables this feature ([#147008](https://github.com/pytorch/pytorch/pull/147008)).
```
Comment (janeyx99) [inline 2.7.0/done/result_inductor.md]:
```suggestion
 - Support fusion of pointwise ops into Template Prologues. `TORCHINDUCTOR_PROLOGUE_FUSION` enables this feature ([#147008](https://github.com/pytorch/pytorch/pull/147008)).
```

File: 2.7.0/done/result_inductor.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+ - Emit a CMakeLists.txt when `package_cpp_only` is specified in AOTI ([#143352](https://github.com/pytorch/pytorch/pull/143352)).
+ - One Dynamo graph can now map to multiple inductor graphs with different `graph_partition` functions. Set the `graph_partition` in inductor config to enable ([#147038](https://github.com/pytorch/pytorch/pull/147038)).
+
+### improvements
+
+ - Add profiling support for codegened CPU FlexAttention kernels ([#145894](https://github.com/pytorch/pytorch/pull/145894)).
+ - Other FlexAttention improvements: ([#147765](https://github.com/pytorch/pytorch/pull/147765)) ([#147435](https://github.com/pytorch/pytorch/pull/147435)) ([#147010](https://github.com/pytorch/pytorch/pull/147010)) ([#146657](https://github.com/pytorch/pytorch/pull/146657)) ([#145059](https://github.com/pytorch/pytorch/pull/145059)) ([#144938](https://github.com/pytorch/pytorch/pull/144938)) ([#143299](https://github.com/pytorch/pytorch/pull/143299)) ([#142281](https://github.com/pytorch/pytorch/pull/142281)) ([#147918](https://github.com/pytorch/pytorch/pull/147918)) ([#148857](https://github.com/pytorch/pytorch/pull/148857)).
```
Comment (janeyx99) [inline 2.7.0/done/result_inductor.md:40]: A bit vague, do you have a better header for this haha

Comment (janeyx99) [reply 2.7.0/done/result_inductor.md:40]: also cc @drisspg

File: 2.7.0/done/result_inductor.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+ - New environment variables to filter cutlass kernels: `TORCHINDUCTOR_CUTLASS_ALLOWLIST` and `TORCHINDUCTOR_CUTLASS_DENYLIST` ([#148161](https://github.com/pytorch/pytorch/pull/148161)).
+ - Add option to disable runtime scalar assertions: `TORCHINDUCTOR_SCALAR_ASSERTS` ([#146462](https://github.com/pytorch/pytorch/pull/146462)).
+ - Add new inductor configs to compiler bisector: `layout_optimization` and `comprehensive_padding` ([#148450](https://github.com/pytorch/pytorch/pull/148450)).
+ - Add an option to skip optimizing generated wrapper code. Set `AOT_INDUCTOR_COMPILE_WRAPPER_WITH_O0=1` ([#144866](https://github.com/pytorch/pytorch/pull/144866)).
+ - Support dynamic shape constraints in Export ([#146044](https://github.com/pytorch/pytorch/pull/146044)).
+ - Handle MLIR scf.yield more accurately in user Triton code ([#147762](https://github.com/pytorch/pytorch/pull/147762)).
+ - Add a global_scratch arg to support Triton 3.3 ([#148051](https://github.com/pytorch/pytorch/pull/148051)).
+ - Removed an unnecessarily struct runtime alignment assertion, allowing more flexible usecases of AOTI ([#143236](https://github.com/pytorch/pytorch/pull/143236)).
```
Comment (janeyx99) [inline 2.7.0/done/result_inductor.md]:
```suggestion
 - Removed an unnecessarily struct runtime alignment assertion, allowing more flexible use cases of AOTI ([#143236](https://github.com/pytorch/pytorch/pull/143236)).
```

File: 2.7.0/done/result_inductor.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+ - Support `_int_mm` in AOTI ([#144571](https://github.com/pytorch/pytorch/pull/144571)).
+ - Support AOTI + CUDAGraphs when calling from Python ([#148601](https://github.com/pytorch/pytorch/pull/148601)).
+ - New post grad pass to remove `torch.ops.aten._assert_tensor_metadata.default` for AOTI ([#145028](https://github.com/pytorch/pytorch/pull/145028)).
+ - Support basic TorchBind in `aot_compile` and `aoti_compile_and_package` ([#148506](https://github.com/pytorch/pytorch/pull/148506)).
+ - Add top level tlparse logging for AOTI ([#147760](https://github.com/pytorch/pytorch/pull/147760))
+
+### bug fixes
+
+ - Fix a bug where the options dictionary on torch.compile calls was ignored ([#145131](https://github.com/pytorch/pytorch/pull/145131)).
+ - Inductor now supports nanj in cpp wrapper CPU ([#144064](https://github.com/pytorch/pytorch/pull/144064)).
```
Comment (janeyx99) [inline 2.7.0/done/result_inductor.md]:
```suggestion
 - Inductor now supports `nanj` in cpp wrapper CPU ([#144064](https://github.com/pytorch/pytorch/pull/144064)).
```

File: 2.7.0/done/result_inductor.md [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+ - Removed an unnecessarily struct runtime alignment assertion, allowing more flexible usecases of AOTI ([#143236](https://github.com/pytorch/pytorch/pull/143236)).
+ - Support `_int_mm` in AOTI ([#144571](https://github.com/pytorch/pytorch/pull/144571)).
+ - Support AOTI + CUDAGraphs when calling from Python ([#148601](https://github.com/pytorch/pytorch/pull/148601)).
+ - New post grad pass to remove `torch.ops.aten._assert_tensor_metadata.default` for AOTI ([#145028](https://github.com/pytorch/pytorch/pull/145028)).
+ - Support basic TorchBind in `aot_compile` and `aoti_compile_and_package` ([#148506](https://github.com/pytorch/pytorch/pull/148506)).
+ - Add top level tlparse logging for AOTI ([#147760](https://github.com/pytorch/pytorch/pull/147760))
+
+### bug fixes
+
+ - Fix a bug where the options dictionary on torch.compile calls was ignored ([#145131](https://github.com/pytorch/pytorch/pull/145131)).
```
Comment (janeyx99) [inline 2.7.0/done/result_inductor.md]:
```suggestion
 - Fix a bug where the options dictionary on `torch.compile` calls was ignored ([#145131](https://github.com/pytorch/pytorch/pull/145131)).
```

File: 2.7.0/done/result_inductor.md [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+ - Inductor now supports nanj in cpp wrapper CPU ([#144064](https://github.com/pytorch/pytorch/pull/144064)).
+ - Fix a bug in the `fractional_max_pool` lowering in Inductor ([#144395](https://github.com/pytorch/pytorch/pull/144395)).
+ - FlexAttention: Fix a few more symbolic shape issues ([#142816](https://github.com/pytorch/pytorch/pull/142816)).
+ - Fix a bug in `associative_scan` ([#143048](https://github.com/pytorch/pytorch/pull/143048)).
+ - Fix the Index Put lowering with same input of self and values ([#139366](https://github.com/pytorch/pytorch/pull/139366)).
+ - Fix a bug in `torch.polygamma(n)` when n == 0 ([#144058](https://github.com/pytorch/pytorch/pull/144058)).
+ - Fix bug in integer `avg_pool` that was causing 0 rounding ([#144059](https://github.com/pytorch/pytorch/pull/144059)).
+ - Change `avg_pool` with `uint` to match eager ([#144313](https://github.com/pytorch/pytorch/pull/144313)).
+ - Fix bug in max-autotune on smaller GPUs (<68 SMs) ([#145133](https://github.com/pytorch/pytorch/pull/145133)).
+ - Fix bug in torch.logit decomposition ([#145576](https://github.com/pytorch/pytorch/pull/145576)).
```
Comment (janeyx99) [inline 2.7.0/done/result_inductor.md]:
```suggestion
 - Fix bug in `torch.logit` decomposition ([#145576](https://github.com/pytorch/pytorch/pull/145576)).
```

File: 2.7.0/done/result_inductor.md [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+ - Make sure not using cpp wrapper when setting nvtx training annotation ([#145538](https://github.com/pytorch/pytorch/pull/145538)).
+ - Fix bug where `SVE256` features were run on `SVE128` systems ([#146207](https://github.com/pytorch/pytorch/pull/146207)).
+ - Fix an unaligned memory access issue in `mm_template` ([#146293](https://github.com/pytorch/pytorch/pull/146293)).
+ - Fix intermediate debug information with `cpp_wrapper` ([#145527](https://github.com/pytorch/pytorch/pull/145527)).
+ - Fix bug where inductor was codegen-ing wrong shapes for bucketize when it was fused as an epilogue ([#148769](https://github.com/pytorch/pytorch/pull/148769)).
+ - Fix bug in AOTI one-pass codegen when max-autotune is turned on ([#143098](https://github.com/pytorch/pytorch/pull/143098)).
+ - Fix a memory leak in package `AOTIModelPackageLoaderPybind::boxed_run` ([#146100](https://github.com/pytorch/pytorch/pull/146100)).
+ - Fix None and equal_to_1 arguments issue in Triton kernel generated by AOTI ([#148102](https://github.com/pytorch/pytorch/pull/148102))
```
Comment (janeyx99) [inline 2.7.0/done/result_inductor.md]:
```suggestion
 - Fix `None` and `equal_to_1` arguments issue in Triton kernel generated by AOTI ([#148102](https://github.com/pytorch/pytorch/pull/148102))
```

File: 2.7.0/done/result_inductor.md [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+ - Update AOTI tutorial ([#143390](https://github.com/pytorch/pytorch/pull/143390)).
+ - `inductor.config.descriptive_names = False` is no longer a suggested option ([#145523](https://github.com/pytorch/pytorch/pull/145523)).
+
+### developers
+
+ - Support for export to unwrap/wrap subclasses AOT, resolves UX issue in torchao where users had to manually unwrap their subclasses before calling export ([#141941](https://github.com/pytorch/pytorch/pull/141941)).
+ - Autotuning logs will now show up in `TORCH_LOG`s under the name "autotuning" ([#147222](https://github.com/pytorch/pytorch/pull/147222)).
+ - Replace `set` by `OrderedSet`: only use OrderedSet in the Inductor codebase ([#138466](https://github.com/pytorch/pytorch/pull/138466)).
+ - Now MPS is considered a `GPU_TYPE` ([#143634](https://github.com/pytorch/pytorch/pull/143634)).
+ - Separate unary post op fusion and lowering for qlinear ([#143903](https://github.com/pytorch/pytorch/pull/143903)).
```
Comment (janeyx99) [inline 2.7.0/done/result_inductor.md]:
```suggestion
 - Separate unary post op fusion and lowering for `qlinear` ([#143903](https://github.com/pytorch/pytorch/pull/143903)).
```

File: 2.7.0/done/result_inductor.md [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+ - New classes to help with kernel memory analysis in heuristics ([#142026](https://github.com/pytorch/pytorch/pull/142026)).
+ - Move ir_pre_fusion.txt and ir_post_fusion.txt from `TORCH_COMPILE_DEBUG` to TORCH_LOGS. For example, `TORCH_LOGS="+ir_pre_fusion"` ([#147248](https://github.com/pytorch/pytorch/pull/147248)).
+ - Implement deepcopy for AOTICompiledModel ([#145423](https://github.com/pytorch/pytorch/pull/145423))
```
Comment (janeyx99) [inline 2.7.0/done/result_inductor.md]:
```suggestion
 - Implement `deepcopy` for AOTICompiledModel ([#145423](https://github.com/pytorch/pytorch/pull/145423))
```

**Review by janeyx99 — COMMENTED** [review]
Please also delete the original file in todo/

File: 2.7.0/done/result_inductor.md (hunk 4 above)
Comment (exclamaforte) [reply 2.7.0/done/result_inductor.md:23]: yeah, that PR removes Inductor support for ~Triton version < ~2.2.0

File: 2.7.0/done/result_inductor.md (hunk 6 above)
Comment (exclamaforte) [reply 2.7.0/done/result_inductor.md:40]: I previously got sign off on the flex attention notes by Driss and the AOTI notes by Bin btw. That header is not the best, but my reasoning is that people are very interested in Flex Attention but none of those changes are big enough for their own bullet. I'd be fine deleting it too

File: 2.7.0/done/result_inductor.md (hunk 4 above)
Comment (exclamaforte) [reply 2.7.0/done/result_inductor.md:23]: maybe something we should highlight elsewhere too

**Review by janeyx99 — APPROVED** [review]
thank you very much!
